### PR TITLE
Feat/541/hex to dec

### LINKED ISSUE #541: [Explorer] Add switch to show transaction value in decimal or hex
image

## Requirements

Like on etherscan, we'd like to have the option to switch from hex to decimal on the transaction values.

### PR DESCRIPTION
**Pull request by Malak67**
Closes #541 

image
image
image
image

### CODE REVIEW

**Review by VargaElod23 — APPROVED** [review]
LGTM 👍🏼

File: services/explorer/src/pages/TransactionData/TopicDataDisplay.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  const handleClose = (val: string) => {
+    setValue(val);
+    setAnchorEl(null);
+    switch (val) {
+      case toggleValues[0].value:
+        setDecoded(hexValue);
+        break;
+      case toggleValues[1].value:
+        const addressHex = '0x' + hexValue.slice(-40);
```
Comment (VargaElod23) [inline services/explorer/src/pages/TransactionData/TopicDataDisplay.tsx:35]: I just checked and this is the wrong way of doing it. This converts the address to decimal. However, on a log's topics we have only addresses, so all we need to do is to cut the padded zeroes to make the address readable.

File: services/explorer/src/pages/TransactionData/EventDataDisplay.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    setSelected(found.value);
+    switch (newValue) {
+      case toggleValues[0].value:
+        setDecoded(hexValue);
+        break;
+      case toggleValues[1].value:
+        const addressHex = '0x' + hexValue.slice(-40);
+        const addressDecimal = BigInt(addressHex).toString();
+        setDecoded(addressDecimal);
```
Comment (VargaElod23) [inline services/explorer/src/pages/TransactionData/EventDataDisplay.tsx:32]: image

This fails on empty data

**Review by VargaElod23 — CHANGES_REQUESTED** [review]

File: services/explorer/src/pages/TransactionData/TopicDataDisplay.tsx (hunk 1 above)
Comment (VargaElod23) [reply services/explorer/src/pages/TransactionData/TopicDataDisplay.tsx:35]: Example: https://etherscan.io/tx/0x9051b74a7a2be1c071453f6b0ff076ab0173b07bb647927933c292c85a8f7076#eventlog

File: services/explorer/src/pages/TransactionData/EventDataDisplay.tsx (hunk 2 above)
Comment (VargaElod23) [reply services/explorer/src/pages/TransactionData/EventDataDisplay.tsx:32]: Also, this won't work like this as we don't know the primitive that comes in here as base, so we should remove this. The main idea for this was to add a `decimal <> hex switch to the tx values and the block numbers` for debug purposes.

**Review by VargaElod23 — APPROVED** [review]

**Review by rattrap — APPROVED** [review]

**Pull request merged**